docs/db/db-relations.md

```diff
@@ -0,0 +1,191 @@
+
+- `objectId` that is associated to the `object` table, this column will identify which object is associated to which event. An object can have multiple events but must have at least one.
+
+---
+
+The `event_user` contains the associations between events and users. This will contain information on which users are registered to which event. An event can have multiple users and a user can have multiple events, this way being a many to many relationship.
+
+---
+
+The `registration` table contains all objects/activities that need registration. Having the following relations:
+
+- `parentId` that is associated to himself. As said above (in the events), it has the same use.
+- `objectId` that is associated to the object table, this is the object reference to the registrations.
+
+---
+
+The `registration_user` table is a relationship many to many between `registration` and `user`. This table contains all users that are registered to an event.
+
+- `registrationId` a registration can have multiple users.
+- `userId` a user can have multiple registrations.
+
+---
+
+### Objects
+
+Object will be the accumulation of the following tables :
+
+- `object`
+- `object_child`
+- `object_status`
+- `object_type`
+
+This part of the database defines the structure of the content.
+
+The `object` table contains the following relations:
+
+- `referenceId` that is associated to himself, an object can have none or many references. This attribute allows the creation of duplicates and allows each campus to create their own objects.
+- `authorId` that is associated to the user table and this is the author of that object/content.
+
+---
+
+The `object_child` table contains the encapsulation of objects. An object can have multiple children and a child must have one object parent. This table contains the following relationships to the object table:
+
+- `parentId` that associates the parent object to the child object.
+- `childId` that belongs to the parent object.
+
+Both child and parent must have at least one association to the `object` table, and the `object` table can have multiple relations with the `object_child` table.
+
+Example:
+
+Campus `madeira` is a **parent object** of `piscine-go`, therefore the later one is the **child**.
+
+But `piscine-go` can be the **parent object** of all the quest, exams and raids (those being the **child objects**). And so on... creating a finite cycle.
+
+---
+
+### Users
+
+Users will be the accumulation of the following tables:
+
+- `role`
+- `user_role`
+- `user`
+- `group`
+- `group_user`
+- `token`
+- `record`
+- `transaction`
+
+---
+
+The `role` table contains permission roles for each user.
+
+---
+
+The `user_role` contains information on which users are associated to which role. A user can have multiple roles and a role can be associated to multiple users.
+
+---
+
+The `group` table is the link between projects and a group of users. This table contains the following relations:
```

Comment:
projects or raids (as project is specific type of content maybe it can be confusing) - or maybe find a neutral word 🤷🏼‍♀️